content/automate/ManageIQ/Transformation/Infrastructure/VM/rhevm.class/__methods__/checkpoweredon.rb

```diff
@@ -11,13 +11,15 @@ def initialize(handle = $evm)
 
               def main
                 task = @handle.root['service_template_transformation_plan_task']
-                destination_vm = @handle.vmdb(:vm).find_by(:id => task.get_option(:destination_vm_id))
-                destination_ems = destination_vm.ext_management_system
-                destination_vm_sdk = ManageIQ::Automate::Transformation::Infrastructure::VM::RedHat::Utils.new(destination_ems).vm_find_by_name(destination_vm.name)
-                @handle.log(:info, "Status of VM '#{destination_vm.name}': #{destination_vm_sdk.status}")
-                unless destination_vm_sdk.status == OvirtSDK4::VmStatus::UP
-                  @handle.root["ae_result"] = "retry"
-                  @handle.root["ae_retry_interval"] = "15.seconds"
+                if task.get_option(:source_vm_power_state) == 'on'
+                  destination_vm = @handle.vmdb(:vm).find_by(:id => task.get_option(:destination_vm_id))
```

Comment:
@fdupont-redhat Should we protect against the case where we don't find the destination_vm?

Reply:
I don't think so as one of the previous state in the state machine waits for the VM to be in the VMDB.